client.go

```diff
@@ -231,3 +237,107 @@ func (c *Client) sendDroppedResponse(e *Event, message string) {
 	c.transmission.SendResponse(r)
 
 }
+
+// MarkerRequest will be used to add details about a marker before
+// creating it
+type MarkerRequest struct {
+	StartTime int    `json:"start_time,omitempty"`
+	EndTime   int    `json:"end_time,omitempty"`
+	Message   string `json:"message,omitempty"`
+	Type      string `json:"type,omitempty"`
+	URL       string `json:"url,omitempty"`
+}
+
+// MarkerResponse stores the marker details whenever a marker is created, updated, deleted, etc
+type MarkerResponse struct {
+	CreatedAt string `json:"created_at"`
+	UpdatedAt string `json:"updated_at"`
+	StartTime int    `json:"start_time"`
+	Message   string `json:"message"`
+	Type      string `json:"type"`
+	ID        string `json:"id"`
+}
+
+// CreateMarker will create a marker for a dataset using the marker request body
+func CreateMarker(dataset, apiKey string, markerRequest MarkerRequest) (MarkerResponse, error) {
```

Comment:
I wanted to add the methods as receiver methods on the config to leverage the dataset and apikey which simplifies the function signatures and reduces redundancy. I did it this way because, from the looks of it, I would have needed to create a more generic transmission implementation (not sure if that was the intent for the prompt) just to add support for more of the basic endpoints.